samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala

```diff
@@ -211,6 +213,17 @@ class TaskStorageManager(
         val systemStreamPartition = new SystemStreamPartition(systemStream, partition)
         val systemConsumer = storeConsumers(storeName)
         val systemConsumerIterator = new SystemStreamPartitionIterator(systemConsumer, systemStreamPartition)
+
+        // Create a store backup engine to restore from hdfs location
+        val map = new util.HashMap[String, String]
+        // TODO: Change these values to something more generic
+        map.put("rocksdb.checkpointDir", System.getProperty("java.io.tmpdir") + "/checkpoint")
+        map.put("hdfs.backupDir", "/Users/eripan/backupTest")
+        // Assumption that restore stores are always logged
+        map.put("rocksdb.dbDir", TaskStorageManager.getStorePartitionDir(loggedStoreBaseDir, storeName, taskName).getAbsolutePath)
+        val backup = new StoreBackupEngine(new MapConfig(map))
+        backup.restoreBackup()
```

Comment:
Need an if-else here so that you don't need both a changelog restore and HDFS restore?